### ISSUE TITLE
Zephyrus DUO 16 fans and power issues and other suggestions

### ISSUE THREAD

**Issue by alanyu1993** [seerge/g-helper#51]
Hello, I appreciate the work here. It's really what I, and most ROG laptop users, want to replace the bloatware armoury crate (AC) which is sadly a must for the laptop.

My laptop is Zephyrus DUO 16 2022. So far I didn't uninstalled AC and had a simple test on G-helper. The performance mode can be switched well by G-helper. However, the Fans and Power feature is not correct. 

The power limit is 150W total and 90W CPU, which I suppose is what G14 could handle. However, other ROG models can handle differently, higher or lower. Could you please increase the max power limit to 250W total and 130W PL1, which are the total and PL1 limits of the ROG Strix Scar 2023? My DUO 16 2022 has the max limit of 230W total and 90W SPL. The better way is to identify the model, set the max limit correspondingly, and give the detailed PL1/PL2 (intel) or SPL/SPPT/FPPT (AMD) setup. I'm not an IT guy, if it's hard to do just skip it.

The fans curve somehow looks wrong. I'm not sure if it is an x/y scale issue or reading the data wrongly. The y-scale is not linear all the way as the link below shows. The actually CPU fan off point is around 60C, but in the profile it shows around 2200 rpm at ~57C, and there is no fan off point.

Other suggestions are: 
1. maybe add the button of "Optimized" instead of the checkbox "set eco on battery", as in the AC the mode is called "Optimized". 
2. display panel profile switch, aka UHD120hz/FHD240hz and single zone/multi zone miniLED switch for some models, though it's not important to me, as I always use UHD mode.
3. give the same options on keybindings, or could you give a txt file to illustrate how to bind, such as Aura switch by customer? On DUO 16, only M4 (original AC button) can be used for this. There is no M3 button.

Thank you again for the work! Have a nice day.
[https://drive.google.com/file/d/1vH6bGfDNnTS_Igy1hVKSLshLedhWIXZe/view?usp=sharing](url)

**Comment by seerge** [comment]
Hi, cool that you have tested app on such a different device ad DUO 2022 :) I don't have it, so for me it's very difficult to check / and develop new features tho.

**Power limits :** do current (total) and (cpu) power limit sliders work on your device at all ? (i.e. do they actually limit power ?) If so - i can extend ranges

**Fans**: Scales are not linear and that's for a reason.It's not possible (at least not on G14/15) to set fan speed lower than 2000rpm. That's why after OFF (bottom) it goes to 2000RPM immediately. That's how ASUS bios seem to treat settings. Is it different on DUO ?

1. Reason I have a checkbox for "optimized GPU" instead of button - is that next to checkbox I can also select a correct button with green/blue border, to indicate current real mode (i.e.  Eco  or Standard)
2. I don't have that display, so I simply don't know at the moment how to set mini-led zones / etc
3. You need to know an event IDs for each specific key pressed. At the moment app allows you to bind keys that are specific to Asus (i.e. not system, like F1) and are NOT handled by windows (i.e. not brightness or volume).

**Comment by alanyu1993** [comment]
Hello, I unistalled AC and tested a bit more systematically, and several issues are critical.

1. the fan curve seems read and set wrongly. The fan stops at lower temperature. It looks like there is an offest from the factory setup. I need to manually set the fan speed to off instead of below 2200 to stop the fan roughly at the same temperature as AC, but I can't drag the points freely. E.g, I can drag all points to 0 rpm in AC, but Ghelper only allows 0 rpm < 60C. With the default setup of performance mode by Ghelper, the fans keep spinning and stopping at the range of 53~60C,  which is extremely annoying, while by AC, the fans are always 0 rpm at this temperature range.
2. power management (PM) setup: it's totally screwed up. The silent/performance/turbo mode PM read by Ghelper is actually performance/turbo/manual mode by AC, which means the fan spins slower at a higher wattage output, so CPU and GPU runs hotter. The lose of silent mode PM is critical, as the battery life is significantly limited. Some numbers for better understanding: Original combined power and max fan speed from AC is ~125W/3000rpm, 175W/4500rpm, 195W/5400rpm, 230W/6100rpm for silent/performance/turbo/manual mode when plugged in, while Ghelper gives 175W/3000rpm, 195W/4500rpm, 230W/5400rpm for silent/performance/turbo mode.
3. Power setup: It doesn't work correctly. The CPU power slider can control the CPU SPL to some extent, but the system only allows SPL no higher than the factory setup. E.g., under silent mode, which is the factory performance mode PM as illustrated in point 2, the factory SPL is 70W. Even I can slide it to 90W, CPU won't go higher than 70W during the test. The total power slider is CPU + iGPU + IO etc instead of not CPU + dGPU. E.g, AIDA64 FPU only, when I set CPU to 90W and total to 100W on performance mode (which is the factory turbo mode PM), CPU can only pull ~85W. It requires total power > CPU + 20W or so to fully give the juice to CPU; AIDA64 FPU + Furmark both, when I set CPU to 45W and total to 100W for silience mode, the total power is still 175W of the factory PM, instead of the slider setup.
4. M4 key aka ROG AC key: Though it indeed can switch Aura effect or performance profile as what I set in the Ghelper, it still triggers AC reminder asking me if I want to install AC to get full use of the laptop.

Conclusion: it can't replace AC on DUO 16 so far. Maybe else need to test deeper on other models. I also suspect that even on G14, the PM could be read one level higher than the fan curve.

It's a good job anyway. I'm still looking forward to the new release to fix the issues.

Suggestion: 
1. Manual mode should be added, as the original PM from AC has 4 levels on ROG models, and the total power slider doesn't really do what it is designed to. AFAIK, even the manual mode on G14 gives 10W higher total power than turbo mode and allows a higher rpm.
2. fan curve should be refined, and remove the limit of 0rpm < 60C. Furthermore,  when one point is dragged to a higher rpm, it would be nice to trigger points for higher temperature.
3. Is there a way to bypass the AC installation reminder of the M4 key?
